File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if(list1 == null && list2 == null) return null;
        if(list1 != null && list2 == null) return list1;
        if(list1 == null && list2 != null) return list2;

        ListNode head;
        ListNode ptr;
        ListNode l1;
        ListNode l2;

        if(list1.val < list2.val) {
            ptr = list1;
            l1 = ptr.next;
            l2 = list2;
        }
        else {
            ptr = list2;
            l1 = list1;
            l2 = ptr.next;
        }
        head = ptr;
        while(l1 != null && l2 != null) {
            if(l1.val < l2.val) {
                ptr.next = l1;
                l1 = l1.next;
            }
            else {
                ptr.next = l2;
                l2 = l2.next;
            }
            ptr = ptr.next;
        }

        if (l1 != null) {
            ptr.next = l1;
        }
        if(l2 != null) {
            ptr.next = l2;
        }

        return head;

    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return None
        if not head.next:
            return head

        prev = head
        curr = head.next
        after = curr.next
        
        head.next = None
        head = None

        while(after):
            curr.next = prev
            prev = curr
            curr = after
            after = after.next
            
        curr.next = prev
        
        return curr